RecoParticleFlow/PFTracking/interface/PFElecTkProducer.h

```diff
@@ -41,11 +43,22 @@ class ConvBremPFTrackFinder;
  and transform them in PFGsfRecTracks.
 */
 
-class PFElecTkProducer final : public edm::stream::EDProducer<> {
+
+
+class PFElecTkProducer final : public edm::stream::EDProducer<edm::GlobalCache<convbremhelpers::HeavyObjectCache> > {
  public:
   
      ///Constructor
-     explicit PFElecTkProducer(const edm::ParameterSet&);
+  explicit PFElecTkProducer(const edm::ParameterSet&, const convbremhelpers::HeavyObjectCache*);
+
+
+  static std::unique_ptr<convbremhelpers::HeavyObjectCache> 
+    initializeGlobalCache( const edm::ParameterSet& conf ) {
+       return std::unique_ptr<convbremhelpers::HeavyObjectCache>(new convbremhelpers::HeavyObjectCache(conf));
```

Comment:
There is a nice way to actually make this shorter (for your future reference)
``` return std::make_unique<convbremhelpers::HeavyObjectCache>(conf);```

it is a C++14 thing.